src/Http/Session/DatabaseSession.php

```diff
@@ -155,7 +154,9 @@ public function write($id, $data): bool
         /** @var string $pkField */
         $pkField = $this->_table->getPrimaryKey();
         $record[$pkField] = $id;
-        $result = $this->_table->save(new Entity($record));
+        $entityClass = $this->_table->getEntityClass();
+        $entity = new $entityClass($record);
```

Comment:
We could just `$this->_table->newEntity($record)`

Reply:
Then need add "id" to $_accessible in Entity Class for mass assigned using newEntity() or patchEntity()

Reply:
You can do that using the options to newEntity().